release/train_tests/torch/image_classification_factory.py

```diff
@@ -0,0 +1,182 @@
+                / ds_output_summary.get_total_wall_time()
+            )
+            
+            iter_stats = stats_summary.iter_stats
+
+            # TODO: Make this raw data dict easier to access from the iterator.
+            # The only way to access this through public API is a string representation.
+            out[ds_key] = {
+                "throughput": ds_throughput,
+                "iter_stats": {
+                    # Prefetch blocks to the training worker.
+                    "prefetch_block-avg": iter_stats.wait_time.avg(),
+                    "prefetch_block-min": iter_stats.wait_time.min(),
+                    "prefetch_block-max": iter_stats.wait_time.max(),
+                    "prefetch_block-total": iter_stats.wait_time.get(),
+
+                    # Actually fetch the block to the training worker.
+                    "fetch_block-avg": iter_stats.get_time.avg(),
+                    "fetch_block-min": iter_stats.get_time.min(),
+                    "fetch_block-max": iter_stats.get_time.max(),
+                    "fetch_block-total": iter_stats.get_time.get(),
+
+                    # Convert a block to a batch by taking a view of the block.
+                    # (This may also do some operations to combine chunks of data
+                    # to make a contiguous chunk memory.)
+                    "block_to_batch-avg": iter_stats.next_time.avg(),
+                    "block_to_batch-min": iter_stats.next_time.min(),
+                    "block_to_batch-max": iter_stats.next_time.max(),
+                    "block_to_batch-total": iter_stats.next_time.get(),
+
+                    # Convert the block to the user-specified batch format (ex: numpy/pandas).
+                    "format_batch-avg": iter_stats.format_time.avg(),
+                    "format_batch-min": iter_stats.format_time.min(),
+                    "format_batch-max": iter_stats.format_time.max(),
+                    "format_batch-total": iter_stats.format_time.get(),
+
+                    # UDF or default collate function converting numpy array -> torch.Tensor.
+                    "collate-avg": iter_stats.collate_time.avg(),
+                    "collate-min": iter_stats.collate_time.min(),
+                    "collate-max": iter_stats.collate_time.max(),
+                    "collate-total": iter_stats.collate_time.get(),
+
+                    # Default finalize function moves torch.Tensor to the assigned train worker GPU.
+                    "finalize-avg": iter_stats.finalize_batch_time.avg(),
+                    "finalize-min": iter_stats.finalize_batch_time.min(),
+                    "finalize-max": iter_stats.finalize_batch_time.max(),
+                    "finalize-total": iter_stats.finalize_batch_time.get(),
+
+                    # Training blocked time
+                    "time_spent_blocked-avg": iter_stats.block_time.avg(),
+                    "time_spent_blocked-min": iter_stats.block_time.min(),
```

Comment:
We should distinguish "first batch blocked time" vs. all the rest of the batches, since the first batch blocked time includes the dataset pipeline "warmup" time. This warmup time drags the average up a lot.